src/Tasks.UnitTests/AssignCulture_Tests.cs

```diff
@@ -216,6 +216,56 @@ public void PseudoLocalization(string culture)
             Assert.Equal($"MyResource.{culture}.resx", t.AssignedFiles[0].ItemSpec);
             Assert.Equal("MyResource.resx", t.CultureNeutralAssignedFiles[0].ItemSpec);
         }
+
+        /*
+        * Method:   AliasedCulture
+        *
+        * Test that an aliased culture (e.g. zh-CN or zh-TW) which is _not_ returned by CultureInfo.GetCultures(CultureTypes.AllCultures)
+        * on Unix-based systems is still considered valid.
+        * See also https://github.com/dotnet/msbuild/issues/3897 (Cultures aliased by ICU cannot be used for resource localization on non-Windows environments)
+        */
+        [Theory]
+        [InlineData("zh-CN")]
+        [InlineData("zh-TW")]
+        public void AliasedCulture(string culture)
+        {
+            AssignCulture t = new AssignCulture();
+            t.BuildEngine = new MockEngine();
+            ITaskItem i = new TaskItem($"MyResource.{culture}.resx");
+            t.Files = new ITaskItem[] { i };
+            t.Execute();
+
+            Assert.Single(t.AssignedFiles);
+            Assert.Single(t.CultureNeutralAssignedFiles);
+            Assert.Equal(culture, t.AssignedFiles[0].GetMetadata("Culture"));
+            Assert.Equal($"MyResource.{culture}.resx", t.AssignedFiles[0].ItemSpec);
+            Assert.Equal("MyResource.resx", t.CultureNeutralAssignedFiles[0].ItemSpec);
+        }
+
+        /*
+        * Method:   InvalidCulture
+        *
+        * Test for invalid culture (i.e. throwing an exception when using new CultureInfo())
+        * and unknown culture (i.e. a culture not known by the operating system but which can be created with new CultureInfo())
+        */
+        [Theory]
+        [InlineData("\U0001F4A5")]
+        [InlineData("xx")]
+        public void InvalidCulture(string culture)
+        {
+            AssignCulture t = new AssignCulture();
+            t.BuildEngine = new MockEngine();
+            ITaskItem i = new TaskItem($"MyResource.{culture}.resx");
+            t.Files = new ITaskItem[] { i };
+            t.Execute();
+
+            Assert.Single(t.AssignedFiles);
+            Assert.Single(t.CultureNeutralAssignedFiles);
+            Assert.Equal(String.Empty, t.AssignedFiles[0].GetMetadata("Culture"));
```

Comment:
This is the critical difference between this test and the above, right? Can you call it out with a comment?

Reply:
Do you mean that the `Culture` metadata is empty?

Reply:
In general I just mean "Please call out the parts of this test that are different". Here I guess it's not just that Culture is empty but also that the culture-neutral part doesn't get assigned.